ext/pcre/php_pcre.c

```diff
@@ -2156,85 +2148,81 @@ static zend_string *php_pcre_replace_array(HashTable *regex, zval *replace, zend
 
 /* {{{ php_replace_in_subject
  */
-static zend_always_inline zend_string *php_replace_in_subject(zval *regex, zval *replace, zval *subject, size_t limit, size_t *replace_count)
+static zend_always_inline zend_string *php_replace_in_subject(
+	zend_string *regex_str, HashTable *regex_ht,
+	zend_string *replace_str, HashTable *replace_ht,
+	zend_string *subject, size_t limit, size_t *replace_count)
 {
 	zend_string *result;
-	zend_string *subject_str = zval_get_string(subject);
-
-	if (Z_TYPE_P(regex) != IS_ARRAY) {
-		result = php_pcre_replace(Z_STR_P(regex),
-								  subject_str,
-								  ZSTR_VAL(subject_str),
-								  ZSTR_LEN(subject_str),
-								  Z_STR_P(replace),
-								  limit,
-								  replace_count);
-		zend_string_release_ex(subject_str, 0);
+
+	if (regex_str) {
+		ZEND_ASSERT(replace_str != NULL);
```

Comment:
This assert doesn't make sense, the condition is checked like one line before it :D

Reply:
They are not the same? One is the regex and the other is the replace string, may still be unnecessary tho :D

Reply:
Ooops, you're right!